TextEditor/TextEditor.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TextEditor
+{
+    class TextEditor
+    {
+        private string output;
+        public string Output
+        {
+            get
+            {
+                return output;
+            }
+
+            private set
+            {
+                output = value;
+            }
+
+        }
+
+        private string outputToUndo = null;
+        private string buffer = null;
+        public string Buffer
+        {
+            get
+            {
+                return buffer;
+            }
+            private set
+            {
+                buffer = value;
+            }
+
+        }
+
+        public TextEditor()
+        {
+            Output = "Это пример текста, которым по умолчанию инициализируется поле _output объекта класса TextEditor при создании объекта" +
+                " посредством конструктора без параметров";
+        }
+
+        public TextEditor(string initialText)
+        {
+            Output = initialText;
+        }
+
+        public void Insert(string text)
+        {
+            outputToUndo = Output;
+            Output += text;
+        }
+
+        public void Delete()
+        {
+            if (Output.Length == 0)
+            {
+                throw new MyException("It's nothing to delete. String is empty.");
+            }
+            else
+            {
+                outputToUndo = Output;
+                Output = Output.Remove(Output.Length - 1);                
+            }
+
+        }
+
+        public void Copy(int startIndex)
+        {
+            if (Output.Length == 0 || startIndex < 0 || startIndex >= Output.Length)
+            {
+                throw new MyException("It's nothing to copy. There's no any symbols to copy or start index out of text bounds.");
+            }
+            else
+            {
+                buffer = Output.Substring(startIndex);
+            }
+
+        }
+
+        public void Paste()
+        {
+            if (Buffer == null)
+            {
+                throw new MyException("It's nothing to paste. Buffer is empty.");
+            }
+            else
+            {
+                outputToUndo = Output;
+                Output += buffer;
+                Buffer = null;
+            }
+
+        }
+
+
+        public void Undo()
+        {
+            if (outputToUndo == null)
```

Comment:
the most tricky thing is to make not only last change, but all the changes undone one by one. please implement it (will require redesign, please don't store all previous revisions of the output)

Reply:
Suggestion applied